src/main/java/frc/robot/subsystems/ArmRotate.java

```diff
@@ -1,18 +1,133 @@
 package frc.robot.subsystems;
 
+import java.util.function.BooleanSupplier;
+import java.util.function.DoubleSupplier;
+
+import org.littletonrobotics.junction.Logger;
+
+import com.chopshop166.chopshoplib.PersistenceCheck;
 import com.chopshop166.chopshoplib.logging.LoggedSubsystem;
+import com.chopshop166.chopshoplib.motors.SmartMotorController;
 
+import edu.wpi.first.math.controller.ProfiledPIDController;
+import edu.wpi.first.math.trajectory.TrapezoidProfile.State;
+import edu.wpi.first.math.util.Units;
+import edu.wpi.first.units.measure.Angle;
+import edu.wpi.first.wpilibj2.command.Command;
 import frc.robot.maps.subsystems.ArmRotateMap;
+import frc.robot.maps.subsystems.ArmRotateMap.ArmRotatePresets;
 import frc.robot.maps.subsystems.ArmRotateMap.Data;
+import frc.robot.maps.subsystems.ElevatorMap;
+import frc.robot.maps.subsystems.ElevatorMap.ElevatorPresets;
 
 public class ArmRotate extends LoggedSubsystem<Data, ArmRotateMap> {
+    final ProfiledPIDController pid;
+    private final double RAISE_SPEED = 0;
+    private final double LOWER_SPEED = 0;
+    private final double MANUAL_LOWER_SPEED_COEF = 0;
+    private final double SLOW_DOWN_COEF = 0;
+    private final double ZEROING_SPEED = 0;
+    double holdAngle = 0;
+
+    ArmRotatePresets level = ArmRotatePresets.OFF;
 
     public ArmRotate(ArmRotateMap armRotateMap) {
         super(new Data(), armRotateMap);
+        pid = armRotateMap.pid;
+    }
+
+    public Command move(DoubleSupplier rotateSpeed) {
+
+        return run(() -> {
+            double speed = rotateSpeed.getAsDouble();
+            double speedCoef = RAISE_SPEED;
+            if (speed < 0) {
+                speedCoef = MANUAL_LOWER_SPEED_COEF;
+            }
+            if (Math.abs(speed) > 0) {
+                level = ArmRotatePresets.OFF;
+                getData().motor.setpoint = (limits(speed * speedCoef));
+            } else if (level == ArmRotatePresets.OFF) {
```

Comment:
Potential gap in logic - what happens if the speed is 0 and the level isn't OFF?